Gemfile.rails-5.0.0

```diff
@@ -0,0 +1,6 @@
+source "https://rubygems.org"
+
+gem 'rails', '5.0.0'
```

Comment:
And here I'll disagree, in example that you've mentioned same redundancy is present
https://github.com/Shopify/active_fulfillment

Using Gemspec for gem publishing, and other Gemspec.*.files with overrides for CI matrix, sounds like solid approach.

This very PR is a bug report about dependencies, which escalated to need for CI matrix, and to usage of different approaches of building CI matrices, and splitting responsibilities between gemspec and Gemfile. I just don't want to have same debate, when rails 5.1 would be announced, if you know what I mean ;)